src/student/commands/list-student-schedules.command.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { pagination } from 'src/common/pagination';
+import { PrismaService } from 'src/database/prisma.service';
+import { SchedulesDto } from '../dto/schedules.dto';
+
+@Injectable()
+export class ListStudentSchedulesCommand {
+  constructor(private prisma: PrismaService) {}
+
+  async execute(user_id: number, query: SchedulesDto): Promise<any> {
+    const studentInclude = {
+      include: {
+        course: true,
+        user: {
+          select: { name: true },
+        },
+      },
+    };
+
+    const monitor = await this.prisma.monitor.findFirst({
+      where: {
+        student_id: user_id,
+      },
+      include: {
+        student: studentInclude,
+      },
+    });
+
+    let orStatement = [{ student_id: user_id }, { monitor_id: monitor?.id }];
+    if (query.eventType === 'monitor') {
+      orStatement = [{ monitor_id: monitor?.id }];
+    } else if (query.eventType === 'student') {
+      orStatement = [{ student_id: user_id }];
+    }
+
+    const schedules = await this.prisma.scheduleMonitoring.findMany({
+      where: {
+        OR: orStatement,
+        AND: [
+          { id_status: Number(query.status) || undefined },
+          { start: { gte: new Date().toISOString() } },
```

Comment:
Essa comparação tá considerando a hora do `new Date()` então o seguinte cenário deu erro:
1. Tenho um agendamento às 18:00 de hoje.
2. Agora são 17:00 e o agendamento está sendo mostrado
3. Se eu mudar o horário de início do agendamento para 15:00 por exemplo, ele some.

Problema: Todos os agendamento do dia atual deveria ser mostrados, mesmo os que já passaram da hora de início.

Reply:
Fixed